### PR TITLE
Create OpenAPIV3 schema for metrics-server package and generate package with the schema

### LINKED ISSUE #2803: Create OpenAPIV3 schema for metrics-server in TCE
## Feature Request
Create OpenAPIV3 schema for metrics-server in TCE

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

## What this PR does / why we need it

Create OpenAPIV3 schema for metrics-server package and generate package with the schema

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
Create OpenAPIV3 schema for metrics-server package and generate package with the schema
```

## Which issue(s) this PR fixes

Fixes: #2803 

## Describe testing done for PR

Tested with ytt commands

$ make generate-openapischema-package PACKAGE=metrics-server VERSION=0.5.1

## Special notes for your reviewer

YTT version needs to be bumped to latest (0.38.0)

### CODE REVIEW

File: addons/packages/metrics-server/0.5.1/package-template.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:yaml", "yaml")
```
Comment (shivaani0505) [inline addons/packages/metrics-server/0.5.1/package-template.yaml:2]: @blc1996 , I was asked to remove the package-template.yaml file from my PR, so we can follow the same trend here

File: addons/packages/metrics-server/0.5.1/bundle/config/schema.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#! schema.yml
+
+#@data/values-schema
+#@schema/desc "OpenAPIv3 Schema for metrics-server"
+---
+#@schema/desc "The namespace in which metrics-server is deployed"
+namespace: kube-system
+metricsServer:
+  #@schema/desc "The namespace in which metrics-server is deployed"
```
Comment (shivaani0505) [inline addons/packages/metrics-server/0.5.1/bundle/config/schema.yaml:9]: Probably, we can modify the description of namespace on L11 to be different from L7 namespace description
Refer to something in kapp controller field `kappController.namespace` : "The namespace value used by older templates, will overwrite will top level namespace of present, keep for backward compatibility" (https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kapp-controller/0.30.0/README.md)

File: addons/packages/metrics-server/0.5.1/bundle/config/schema.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#! schema.yml
```
Comment (shivaani0505) [inline addons/packages/metrics-server/0.5.1/bundle/config/schema.yaml:1]: should be changed to schema.yaml

**Comment by shivaani0505** [comment]
@blc1996 you need to create a separate issue in tce and fix "Which issue(s) this PR fixes" section, Issue mentioned there is for Secret Gen controller

File: addons/packages/metrics-server/0.5.1/package-template.yaml (hunk 1 above)
Comment (blc1996) [reply addons/packages/metrics-server/0.5.1/package-template.yaml]: removed

File: addons/packages/metrics-server/0.5.1/bundle/config/schema.yaml (hunk 2 above)
Comment (blc1996) [reply addons/packages/metrics-server/0.5.1/bundle/config/schema.yaml]: Changed

**Comment by blc1996** [comment]
Hi @ggpaue , could you take a look at this MR?

**Comment by ggpaue** [comment]
@blc1996 the change looks good to me and I will approve it, can you add what ytt commands you used for testing in the description before merge? thanks

**Review by ggpaue — APPROVED** [review]
lgtm

**Pull request merged**